From d5ce1cc17bb5bdb78b5144731098d4fe54333335 Mon Sep 17 00:00:00 2001
From: Rakesh Ranjan Jha <rakesh.r.jha@intel.com>
Date: Mon, 16 Nov 2015 18:31:08 +0530
Subject: [PATCH] Disable V1P2A rail conditionally

VIP2A rail is dedicated on FFD designs for CHT-HR/T4 but is shared
with multiple consumers (e.g. SoC I/Os, UFS, Modem, eMMC connector)
on RVP designs with WC PMIC. So this rail needs to be controlled for
FFD designs but needs to be always enabled for RVP. Therefore added a
check for the same.

Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9676
Change-Id: I6d8e3d63590a568fab13594e51d94a91b1507bd9
Signed-off-by: Rakesh Ranjan Jha <rakesh.r.jha@intel.com>
---
 drivers/mfd/intel_soc_pmic_wc.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/intel_soc_pmic_wc.c b/drivers/mfd/intel_soc_pmic_wc.c
index 89dec70..b6e1e8e 100644
--- a/drivers/mfd/intel_soc_pmic_wc.c
+++ b/drivers/mfd/intel_soc_pmic_wc.c
@@ -46,6 +46,7 @@
 #include <linux/regulator/driver.h>
 #include <linux/regulator/gpio-regulator.h>
 #include <linux/platform_device.h>
+#include <linux/dmi.h>
 
 
 #define CHIPID		0x00
@@ -893,6 +894,7 @@ static void wcove_set_bcu_pdata(void)
 
 static int whiskey_cove_init(void)
 {
+	const char *board_name = dmi_get_system_info(DMI_BOARD_NAME);
 	pr_info("Whiskey Cove: ID 0x%02X, VERSION 0x%02X\n",
 		intel_soc_pmic_readb(CHIPID), intel_soc_pmic_readb(CHIPVER));
 
@@ -900,7 +902,14 @@ static int whiskey_cove_init(void)
 	wcove_set_bcu_pdata();
 	wc_set_adc_pdata();
 	wc_set_gpio_pdata();
-	wc_set_v1p2_pdata();
+
+	/* V1P2A is shared with multiple consumer on RVP designs
+	* and is required to be always enabled so skip for RVP
+	*/
+
+	if (board_name && !strcmp(board_name, "Cherry Trail FFD"))
+		wc_set_v1p2_pdata();
+
 	wc_set_v1p8_pdata();
 	wc_set_v2p8_pdata();
 	wc_set_vprog4b_pdata();
-- 
1.9.1

